Add dna-writer support for TXR4 peripherals

emusan opened the pull request.

Commit:
Add dna-writer support for TXR4 peripherals

janovetz commented on dna-tools/dna-writer.cpp:
```diff
@@ -72,10 +72,15 @@ int main(int argc, char *argv[])
 	// work.
-	dna_header[SZG_DNA_PTR_DNA_REQUIRED_MAJOR] = 0;
-	dna_header[SZG_DNA_PTR_DNA_REQUIRED_MINOR] = 0;
+	if (dna_json["is_txr4"].get<bool>() == true) {
```
For this sort of thing, a progressive test might work better...

1. Setup the sensible default.
2. If (condition1) then add a new requirement.
3. If (condition2) then add a new requirement.

Things get complicated to cover all the tasks, so we can just go with what you have here and make it more complex later.

1. Setup the sensible default.
2. If (condition1) then assure rev >= 1.1
3. If (condition2) then assure rev >= 1.4
4. If condition3) then assure rev > 1.2
5. etc.

emusan replied:
I've updated the method I'm using for this in the latest commit (4c359044fd), if that looks good to you I'll merge this all.

janovetz replied:
Looks good to me.

Commit:
Check peripheral minimum required version

Commit:
Add additional test for required minor version

Commit:
Add support for TXR4 detection

Commit:
Merge pull request #3 from SYZYGYfpga/smartvio-txr4

Add support for TXR4 detection

Commit:
Merge pull request #2 from SYZYGYfpga/smartvio-reqver

Check peripheral minimum required version

Commit:
Update DNAWriter version criteria based on review

Move to a progressive test rather than what will be a large if/else
statement as this grows in the future.
